### PR TITLE
Fixes #6781: Display node compliances in Nodes lists

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/6781

### CODE REVIEW

File: rudder-web/src/main/scala/com/normation/rudder/web/services/SrvGrid.scala [hunk 1]
```diff
@@ -132,8 +136,95 @@ class SrvGrid(
...
+            val error = eb ?~! "error while fetching compliances"
+            logger.error(error.messageChain)
+            Alert(error.messageChain)
+        }
+      } else {
+        After(500,ajaxCompliance(tableId,future))
+      }
+    } )
+
+  }
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/services/SrvGrid.scala]: That code and the other belove are almost line for line what is in RuleGrid. Could you factor out that code and use it in the two grid - else maintenance will be horrible.

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
PR updated

File: rudder-web/src/main/scala/com/normation/rudder/web/services/AsyncComplianceService.scala [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    def empty : Boolean = nodeIds.isEmpty
+    def groupBy(report : RuleNodeStatusReport) : NodeId = report.nodeId
+  }
+
+  private[this] class RuleCompliance(
+      val nodeIds: Set[NodeId]
+    , val ruleIds: Set[RuleId]
+  ) extends ComplianceBy[RuleId] {
+    def value(key : RuleId) : String = key.value
+    val jsContainer : String = "nodeCompliances"
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/services/AsyncComplianceService.scala]: shouldn't it be "ruleCompliance" ?

Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/services/AsyncComplianceService.scala]: ruleCompliances (with 's')

**Comment by VinceMacBuche** [comment]
PR updated

**Pull request merged**